core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -624,6 +624,47 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(store.getRemoteBytes("list1").isEmpty)
   }
 
+  Seq(0, Int.MaxValue - 512).foreach { maxRemoteBlockSizeFetchToMem =>
+    // Trying different values of maxRemoteBlockSizeFetchToMem to test both
+    // non-streaming and streaming flows of Block replication
+    test(s"test for Block replcation retry logic with " +
+      s"maxRemoteBlockSizeFetchToMem: ${maxRemoteBlockSizeFetchToMem}") {
+      val storageLevel = StorageLevel(
+        useDisk = false, useMemory = true, deserialized = true, replication = 3)
+      // Retry replication logic for 2 failures
+      conf.set("spark.storage.maxReplicationFailures", "2")
```

Comment:
do we have config entry of it in `org.apache.spark.internal.config`?

Reply:
Yes there is. Replaced all config strings with Config entry in this PR.